### PR TITLE
[FEATURE] [MER-3807] My assignments student view

### PR DESCRIPTION
**Pull request by nicocirio**
[Link](https://eliterate.atlassian.net/browse/MER-3807) to the ticket

- [X] Light and dark mode
- [X] Back button to return to assignments
- [X] Responsive (added ellipsis when we have an overflow on the title)
- [X] Reduced memory footprint

https://github.com/user-attachments/assets/6f83c068-be96-46a2-99c0-cf57d41293af

### CODE REVIEW

File: lib/oli_web/live/delivery/student/assignments_live.ex [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      )
+    end)
+  end
+
+  def handle_info(:gc, socket) do
+    # manually garbage collect to reduce memory usage after mount/3
+    :erlang.garbage_collect(socket.transport_pid)
+    :erlang.garbage_collect(self())
+    {:noreply, socket}
+  end
```
Comment (gastonabella) [inline lib/oli_web/live/delivery/student/assignments_live.ex:61]: Since this piece of code is starting to repeat across different LiveViews, I'm beginning to think that it might be a good idea to extract it to the `OliWeb` module as an option of the `live_view` macro.

Something like
```elixir
use OliWeb, :live_view, garbage_collect: true
```

We could discuss it further and do the refactor in a separate PR. I'm just opening the discussion :)

**Review by gastonabella — APPROVED** [review]
Great work!

File: lib/oli_web/live/delivery/student/assignments_live.ex (hunk 1 above)
Comment (nicocirio) [reply lib/oli_web/live/delivery/student/assignments_live.ex:61]: I love the idea! 🚀
